### PR TITLE
Optimize collectibles DB query

### PR DESCRIPTION
**Pull request by Uxio0**
- Use a proper query instead of returning all elements and do the processing on Python
- Related to https://github.com/safe-global/safe-pm/issues/89

### CODE REVIEW

File: safe_transaction_service/history/models.py [hunk 1]
```diff
@@ -499,41 +500,45 @@ def to_erc721_transfer(self):
...
+                                WHERE  "_from" = %s AND "to" != "_from"
+                                GROUP  BY address,
+                                          token_id) Q2
+                            ON Q1.address = Q2.address
+                               AND Q1.token_id = Q2.token_id
+                WHERE Q1.count > COALESCE(Q2.count, 0)
+                """,
+                [hex_address, hex_address],
+            )
+            return [
```
Comment (moisses89) [inline safe_transaction_service/history/models.py:538]: What would it happen if there is any change on the database model? Would be detected by unit test to change the query acordingly?

Comment (Uxio0) [reply safe_transaction_service/history/models.py:538]: Yes

Comment (moisses89) [reply safe_transaction_service/history/models.py:538]: Ok, I would like to suggest catch the exception (if model changes) and mange it using orm queries like a second chance. What do you think?

Comment (Uxio0) [reply safe_transaction_service/history/models.py:538]: I don't agree. If model changes we will notice due to the tests. If we do as you propose and model changes, tests will keep passing and we will not notice that we are using a suboptimal version of the query

Comment (Uxio0) [reply safe_transaction_service/history/models.py:538]: I'm not very fond of using raw queries unless it's totally needed, but I'm alright if they are well tested

Comment (moisses89) [reply safe_transaction_service/history/models.py:538]: Make sense, I hope that unit test get the change before integration test.

**Review by moisses89 — APPROVED** [review]

File: safe_transaction_service/history/models.py (hunk 1 above)
Comment (Uxio0) [reply safe_transaction_service/history/models.py:538]: Of course. You can try as you don't seem convinced 😉

**Pull request merged**